index.js

```diff
@@ -134,67 +136,26 @@ function select(cb) {
 
 function stream() {
   let buff = ''
-  let len = 0
-  let depth = 0
-  let isString = false
 
   let count = 0
-  let head = ''
-  const check = (i) => {
-    if (depth <= 0) {
-      const input = buff.substring(0, len + i + 1)
-
-      if (count > 0) {
-        if (head !== '') {
-          const json = JSON.parse(head)
-          apply(json)
-          head = ''
-        }
-
-        const json = JSON.parse(input)
-        apply(json)
-      } else {
-        head = input
-      }
-
-      buff = buff.substring(len + i + 1)
-      len = buff.length
-      count++
-    }
-  }
 
   return {
     isStream() {
       return count > 1
     },
     value() {
-      return head + buff
+      return buff
     },
     read() {
       let chunk
-
       while ((chunk = stdin.read())) {
-        len = buff.length
         buff += chunk
-
-        for (let i = 0; i < chunk.length; i++) {
-          if (isString) {
-            if (chunk[i] === '"') {
-              if ((i === 0 && buff[len - 1] !== '\\') || (i > 0 && chunk[i - 1] !== '\\')) {
-                isString = false
-                check(i)
-              }
-            }
-            continue
-          }
-
-          if (chunk[i] === '{' || chunk[i] === '[') {
-            depth++
-          } else if (chunk[i] === '}' || chunk[i] === ']') {
-            depth--
-            check(i)
-          } else if (chunk[i] === '"') {
-            isString = true
+        let parts = buff.split(os.EOL);
```

Comment:
This will trigger apply for every check for all previous messages as well.

Reply:
No, buff turns into the last part of parts because of `buff = parts.pop()`

Reply:
But I think be able to consume any json stream is better as we can use for example `cat *.json`